lawnchair/src/app/lawnchair/ui/preferences/CustomIconShapeCreator.kt

```diff
@@ -0,0 +1,323 @@
+package app.lawnchair.ui.preferences
+
+import android.graphics.PointF
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.requiredWidthIn
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.Icon
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.RadioButton
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.ArrowDropDown
+import androidx.compose.material3.Button
+import androidx.compose.material3.LocalContentColor
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Slider
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavGraphBuilder
+import app.lawnchair.icons.shape.IconCornerShape
+import app.lawnchair.icons.shape.IconShape
+import app.lawnchair.preferences.getAdapter
+import app.lawnchair.preferences2.preferenceManager2
+import app.lawnchair.ui.AlertBottomSheetContent
+import app.lawnchair.ui.preferences.components.BottomSpacer
+import app.lawnchair.ui.preferences.components.IconShapePreview
+import app.lawnchair.ui.preferences.components.PreferenceDivider
+import app.lawnchair.ui.preferences.components.PreferenceGroup
+import app.lawnchair.ui.preferences.components.PreferenceLayout
+import app.lawnchair.ui.preferences.components.PreferenceTemplate
+import app.lawnchair.ui.preferences.components.getSteps
+import app.lawnchair.ui.preferences.components.snapSliderValue
+import app.lawnchair.ui.util.LocalBottomSheetHandler
+import com.android.launcher3.R
+import kotlin.math.roundToInt
+
+fun NavGraphBuilder.customIconShapeCreatorGraph(route: String) {
+    preferenceGraph(route, { CustomIconShapeCreator() })
+}
+
+@Composable
+private fun CustomIconShapeCreator() {
```

Comment:
Maybe change the name to `CustomIconShapePreference` to match the other Preference files? This applies to other parts of this PR.

You could also move the custom icon creator itself (and the parts of it) to the `/components/` folder and simply call it in the preference, similar to the `IconShapePreference` implementation.

Reply:
Good idea! I'll apply this change now.